src/libraries/System.Private.CoreLib/src/System/Decimal.DecCalc.cs

```diff
@@ -1687,173 +1688,41 @@ internal static void VarDecFromR4(float input, out DecCalc result)
             /// <summary>
             /// Convert double to Decimal
             /// </summary>
+
             internal static void VarDecFromR8(double input, out DecCalc result)
             {
                 result = default;
 
-                // The most we can scale by is 10^28, which is just slightly more
-                // than 2^93.  So a float with an exponent of -94 could just
+                // The smallest non-zero decimal we can represent is 10^-28, which is just slightly more
+                // than 2^-93.  So a float with an exponent of -94 could just
                 // barely reach 0.5, but smaller exponents will always round to zero.
+                // This is a shortcut to catch doubles that are too small efficiently.
                 //
-                const uint DBLBIAS = 1022;
-                int exp = (int)(GetExponent(input) - DBLBIAS);
-                if (exp < -94)
-                    return; // result should be zeroed out
-
-                if (exp > 96)
-                    Number.ThrowOverflowException(TypeCode.Decimal);
-
-                uint flags = 0;
-                if (input < 0)
-                {
-                    input = -input;
-                    flags = SignMask;
-                }
-
-                // Round the input to a 15-digit integer.  The R8 format has
-                // only 15 digits of precision, and we want to keep garbage digits
-                // out of the Decimal were making.
-                //
-                // Calculate max power of 10 input value could have by multiplying
-                // the exponent by log10(2).  Using scaled integer multiplcation,
-                // log10(2) * 2 ^ 16 = .30103 * 65536 = 19728.3.
-                //
-                double dbl = input;
-                int power = 14 - ((exp * 19728) >> 16);
-                // power is between -14 and 43
-
-                if (power >= 0)
+                if (input.Exponent < -94)
```

Comment:
I'm using the `Exponent` exposed on `double` now. I think this is more correct than the original comparison.